examples/stable_diffusion_v2/ldm/models/diffusion/ddim.py

```diff
@@ -356,11 +355,15 @@ def encode(
             alphas_next = self.ddim_alphas[:num_steps]
             alphas = self.ddim_alphas_prev[:num_steps]
 
+        timesteps = np.arange(self.ddpm_num_timesteps) if use_original_steps else self.ddim_timesteps
+        timesteps = timesteps[:num_steps]
+        iterator = tqdm(timesteps, desc="Encoding image", total=timesteps.shape[0])
+
         x_next = x0
         intermediates = []
         inter_steps = []
```

Comment:
is `timesteps` and `range(num_steps)` yield the same result in all situations

Reply:
No. They are not yielding the same results unless `timesteps` is the same as `range(num_steps)`.

Actually, I think there was a bug before I edited it. I take the decode func ([line 421-432](https://github.com/mindspore-lab/mindone/blob/5d2555df652f0ff7d22644f8a38709f7598182c0/examples/stable_diffusion_v2/ldm/models/diffusion/ddim.py#L421)) as a reference and revised the encode func. The major difference is that decode func flips the timesteps, while the encode func does not.